configuration/src/secrets.rs

```diff
@@ -36,17 +36,22 @@ impl AgentSecrets {
 
         for network in networks.iter() {
             let network_upper = network.to_uppercase();
-            let chain_conf = ChainConf::from_env(&format!("RPCS_{}", network_upper))?;
-            let transaction_signer =
-                SignerConf::from_env(&format!("TRANSACTIONSIGNERS_{}", network_upper))?;
+
+            let chain_conf =
+                ChainConf::from_env(&format!("RPCS_{}", network_upper), Some("RPCS_DEFAULT"))?;
+
+            let transaction_signer = SignerConf::from_env(
+                &format!("TRANSACTIONSIGNERS_{}", network_upper),
+                Some("TRANSACTIONSIGNERS_DEFAULT"),
+            )?;
```

Comment:
I think they should be kept prefix-agnostic. You can call `SignerConf::from_env` a second time with the default prefix but `ChainConf::from_env` needs to be able to handle two prefixes (`*_CONNECTION_URL` has no default). Passing prefix and default prefix separately gives `FromEnv` implementers the required flexibility.

Reply:
Maybe you're right about this. Smells to me for some reason but also simplifies.

Reply:
The other option is to separate prefix and network and have network be an optional param. You can first check DEFAULT_{prefix} then check {network}_{prefix} or {prefix}_{network} this way. Think this may be more intuitive than using a hardcoded static string as param

Reply:
Actually, I'm going to stick with the original. I think it's ok for an impl to have hardcoded keys (makes little sense to pass everything in) but the interface should indicate that a default is going to be checked. Passing in only network/prefix and then having it pull a default value is too much magic imo. The alternative would be a `check_default` param or sth similar but I think an `Option<&str>` is fine.

Reply:
Expected precedence from lowest to highest:
- DEFAULT_{prefix}